File: cost(update).py
# ad multiple food at time
def calculate_total_cost(food_list):
    total_cost = 0
    for food in food_list:
        food_type, number_of_items, cost_per_item = food
        total_cost += number_of_items * cost_per_item
    return total_cost


# get input from the user
food_list = []
for i in range(20):
    food_type = input("Enter the type of food (" + str(i + 1) + " of 20): ")
    number_of_items = input("Enter the number of items: ")
    cost_per_item = input("Enter the cost per item: ")
    try:
        number_of_items = float(number_of_items)
        cost_per_item = float(cost_per_item)
    except ValueError:
        print("Invalid input. Please enter valid numbers.")
    food_list.append((food_type, number_of_items, cost_per_item))

# calculate the total cost
total_cost = calculate_total_cost(food_list)

# print the result
print("The total cost of your purchase is: $" + str(total_cost))

File: cost(update).java
import java.util.Scanner;

public class FoodCostCalculator {
  public static void main(String[] args) {
    // Create a new Scanner object to read input from the user
    Scanner scanner = new Scanner(System.in);

    // Create a variable to store the total cost of all food items in KES
    int totalCost = 0;

    // Loop until the user enters "done"
    while (true) {
      // Prompt the user for the name of the food item
      System.out.print("Enter the name of the food item (or 'done' to exit): ");
      String foodItem = scanner.nextLine();

      // If the user entered "done", exit the loop
      if (foodItem.equals("done")) {
        break;
      }

      // Prompt the user for the number of items
      System.out.print("Enter the number of items: ");
      int numItems = scanner.nextInt();

      // Prompt the user for the cost per item in KES
      System.out.print("Enter the cost per item in KES: ");
      int costPerItem = scanner.nextInt();

      // Calculate the total cost for this food item in KES
      int itemCost = numItems * costPerItem;

      // Add the total cost for this food item to the overall total cost in KES
      totalCost += itemCost;

      // Print the total cost for this food item in KES
      System.out.println("The total cost for " + numItems + " " + foodItem + " is KES " + itemCost);
    }

    // Print the total cost for all food items in KES
    System.out.println("The total cost for all food items is KES " + totalCost);
  }
}

//This program is similar to the previous example, but it uses Kenyan shillings (KES) as the currency and it uses int variables to store the cost per item and the total cost, since KES is a whole number currency. The program prompts the user for the cost per item in KES and calculates the total cost in KES, and it prints the total cost in KES.